src/libsystemd-network/dhcp-static-lease.c

```diff
@@ -0,0 +1,141 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "dhcp-internal.h"
+#include "dhcp-server-internal.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+#define ADDR_NULL 0
+
+static sd_dhcp_static_lease *current_static_lease = NULL;
+
+static sd_dhcp_static_lease* sd_dhcp_static_lease_free(sd_dhcp_static_lease *i) {
+        if (!i)
+                return NULL;
+
+        free(i->client_id.data);
+
+        return mfree(i);
+}
+
+_destructor_ static void sd_dhcp_static_lease_current_cleanup(void) {
+        sd_dhcp_static_lease_free(current_static_lease);
+        TAKE_PTR(current_static_lease);
+}
+
+static int dhcp_static_lease_new(DHCPClientId *client_id, const struct in_addr *address, sd_dhcp_static_lease **ret) {
+        sd_dhcp_static_lease *p = new(sd_dhcp_static_lease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        assert_return(ret, -EINVAL);
+        assert_return(address, -EINVAL);
+
+        *p = (sd_dhcp_static_lease) {
+                .n_ref = 1,
+                .client_id = *client_id,
+                .address = le32toh(address->s_addr)
+        };
+
+        *ret = TAKE_PTR(p);
+        return 0;
+}
+
+static int dhcp_static_lease_new_default(sd_dhcp_static_lease **ret) {
+        struct in_addr address = { ADDR_NULL };
+        DHCPClientId client_id = { .length = 1, .data = NULL };
```

Comment:
ditto, no need to initialize `.data`, it's default zero-initialized if not specified anyway